k8s/postgresql/apptest/tester/tests/basic-suite.yaml

```diff
@@ -1,8 +1,141 @@
 actions:
-- name: dummy test
+- name: secure login
   bashTest:
     script: |-
-      exit 0
+      SERVICE="${APP_INSTANCE_NAME}-postgresql-svc"
+      IP=''
+      PGPASSWORD=''
+      FAILED=0
+
+      function wait_for_service_external_ip {
+        IP=$(kubectl get svc "${SERVICE}" \
+          --namespace "${NAMESPACE}" \
+          -ojsonpath="{.status.loadBalancer.ingress[0].ip}")
+        if [[ $? -ne 0 ]]; then
+          IP=""
+        fi
+      }
+
+      function get_postgresql_postgres_password {
+        PGPASSWORD=$(kubectl get secret "${APP_INSTANCE_NAME}-secret" \
```

Comment:
I'm not sure why do we need this function - can you elaborate on that?

Reply:
Connection to PostgreSQL requires username and password. This function extracts password from secret.

Reply:
Sure, but why loop? Is it possible that secret will not be set?

Reply:
There are three reasons:
1. Testing system is not reliable. I'm sure it starts tests before all resources are ready.
2. There can be a connection timeout (eg. master node upgrade) during tests.
3. I just copied wait_for_service_external_ip and edited it a bit.

Reply:
Function removed.